camel/Output/OutputEndpointData.php

```diff
@@ -137,7 +137,18 @@ public static function fromExtractedEndpointArray(array $endpoint): OutputEndpoi
 
     public function endpointId(): string
     {
-        return $this->httpMethods[0] . str_replace(['/', '?', '{', '}', ':', '\\', '+', '|'], '-', $this->uri);
+        return sprintf('%s@%s', $this->httpMethods[0], $this->uri);
+    }
+
+    public function cssEscapedEndpointId(): string
```

Comment:
I see what you're doing, but for consistency, why can't we just use the escaped version directly as the normal `endpointId()`? I'd prefer we not have two different types of endpoint IDs.

Reply:
JS DOM finding will break with some characters, such as '@', ':', others, 'cause it might be css modifiers.